src/main/java/ru/ayaz/UserMessageDistributor.java

```diff
@@ -0,0 +1,63 @@
+package ru.ayaz;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import static ru.ayaz.MessageParser.isCommand;
+
+public class UserMessageDistributor implements Runnable {
+
+    private Map<String, User> userMap;
+    private Map<String, UserSocketHandler> userSocketHandlerMap;
+    private BlockingQueue<UserMessage> messageQueue;
+    private MessageBroadcaster broadcaster;
+    private CommandExecutor executor;
+
+    UserMessageDistributor(HashMap<String, User> userMap, HashMap<String, UserSocketHandler> userSocketHandlerMap,
```

Comment:
I mean we implemented Dependency Injection pattern so we can unit test Distributor. 
Maps cant be accessed from anywhere but Chat which creates them.